File: oci-start-mac/OciStart/Features/IpQuality/IpQualityService.swift
import Foundation

/// Network layer for IP quality settings (`/system/ipSettings` · `/api/system/ipSettingsConfigs`).
struct IpQualityService {
    let baseURL: String
    private let client = APIClient.shared

    /// GET `/api/system/ipSettingsConfigs`
    func fetchConfigs() async throws -> IpQualityConfigs {
        let url = try client.makeURL(baseURL, path: "/api/system/ipSettingsConfigs")
        let raw = try await client.getJSON(url)
        return try IpQualityJSON.parseConfigs(raw)
    }

    /// POST `/api/system/updateIpCheckConfig`
    func updateIpCheck(enabled: Bool, checkInterval: Int) async throws {
        let url = try client.makeURL(baseURL, path: "/api/system/updateIpCheckConfig")
        let raw = try await client.postJSON(url, body: [
            "enabled": enabled,
            "checkInterval": checkInterval
        ])
        try IpQualityJSON.ensureOK(raw, fallback: "保存 IP 检测配置失败")
    }

    /// POST `/system/vps/saveConfig`
    func saveVPS(_ config: VPSConfigDTO) async throws {
        let url = try client.makeURL(baseURL, path: "/system/vps/saveConfig")
        let raw = try await client.postJSON(url, body: [
            "type": config.type,
            "enabled": config.enabled,
            "serverIp": config.serverIp,
            "username": config.username,
            "password": config.password,
            "sshPort": config.sshPort
        ])
        try IpQualityJSON.ensureOK(raw, fallback: "保存 VPS 配置失败")
    }

    /// POST `/system/vps/testConnection`
    func testConnection(_ config: VPSConfigDTO) async throws -> String {
        let url = try client.makeURL(baseURL, path: "/system/vps/testConnection")
        let raw = try await client.postJSON(url, body: [
            "type": config.type,
            "serverIp": config.serverIp,
            "username": config.username,
            "password": config.password,
            "sshPort": config.sshPort
        ])
        return try IpQualityJSON.parseTestResult(raw)
    }
}

File: oci-start-win/src/OciStart/Features/IpQuality/IpQualityService.cs
using OciStart.Core;

namespace OciStart.Features.IpQuality;

public sealed class IpQualityService
{
    private readonly ApiClient _api = ApiClient.Shared;

    public async Task<IpQualityConfigs> FetchConfigsAsync(CancellationToken ct = default)
    {
        var raw = await _api.GetJsonAsync("/api/system/ipSettingsConfigs", ct: ct).ConfigureAwait(false);
        return IpQualityJson.ParseConfigs(raw);
    }

    public async Task UpdateIpCheckAsync(bool enabled, int checkInterval, CancellationToken ct = default)
    {
        var raw = await _api.PostJsonAsync("/api/system/updateIpCheckConfig", new
        {
            enabled,
            checkInterval
        }, ct: ct).ConfigureAwait(false);
        ApiClient.EnsureApiOk(raw, "保存 IP 检测配置失败");
    }

    public async Task SaveVpsAsync(VpsConfigDto config, CancellationToken ct = default)
    {
        var raw = await _api.PostJsonAsync("/system/vps/saveConfig", new
        {
            type = config.Type,
            enabled = config.Enabled,
            serverIp = config.ServerIp,
            username = config.Username,
            password = config.Password,
            sshPort = config.SshPort
        }, ct: ct).ConfigureAwait(false);
        ApiClient.EnsureApiOk(raw, "保存 VPS 配置失败");
    }

    public async Task<string> TestConnectionAsync(VpsConfigDto config, CancellationToken ct = default)
    {
        var raw = await _api.PostJsonAsync("/system/vps/testConnection", new
        {
            type = config.Type,
            serverIp = config.ServerIp,
            username = config.Username,
            password = config.Password,
            sshPort = config.SshPort
        }, ct: ct).ConfigureAwait(false);
        return IpQualityJson.ParseTestResult(raw);
    }
}
